daft/dataframe/lance_write_sink.py

```diff
@@ -0,0 +1,126 @@
+import pathlib
+from itertools import chain
+from typing import Iterator, List, Literal, Optional, Union
+
+import lance
+
+from daft.context import get_context
+from daft.daft import IOConfig
+from daft.datatype import DataType
+from daft.io import DataSink, WriteOutput
+from daft.recordbatch import MicroPartition
+from daft.schema import Schema
+
+
+class LanceWriteSink(DataSink[list[lance.FragmentMetadata]]):
+    """WriteSink for writing data to a Lance dataset."""
+
+    def _import_lance(self):
+        try:
+            import lance
+
+            return lance
+        except ImportError:
+            raise ImportError("lance is not installed. Please install lance using `pip install daft[lance]`")
+
+    def __init__(
+        self,
+        uri: Union[str, pathlib.Path],
+        schema: Schema,
```

Comment:
why do we need to pass in a schema? This should be only the case if we are creating a new lance dataset.

Reply:
This was part of the existing checks for `.write_lance()` where we check if the schema of the dataframe matches the existing Lance dataset's schema. And if it wasn't, we would error out early